### PR TITLE
txnbuild: Reinstate missing fee bump transaction functions

### PR DESCRIPTION
**Pull request by tamirms**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

A few functions on the fee bump transaction struct were unintentionally removed in https://github.com/stellar/go/pull/3048/files

This commit reinstates those deleted functions and adds tests for the functions to prevent them from being deleted accidentally again.

### CODE REVIEW

**Review by marcelosalloum — APPROVED** [review]
LGTM!

File: txnbuild/transaction.go [hunk 1]
```diff
@@ -454,6 +460,45 @@ func (t *FeeBumpTransaction) Sign(network string, kps ...*keypair.Full) (*FeeBum
...
+// with an additional signature derived from the given base64-encoded signature.
+func (t *FeeBumpTransaction) AddSignatureBase64(network, publicKey, signature string) (*FeeBumpTransaction, error) {
+	extendedSignatures, err := concatSignatureBase64(t.envelope, t.signatures, network, publicKey, signature)
+	if err != nil {
+		return nil, err
+	}
+
+	newTx := new(FeeBumpTransaction)
+	*newTx = *t
+	newTx.signatures = extendedSignatures
```
Comment (Shaptic) [inline txnbuild/transaction.go:498]: Can I ask what this is doing? Is this like a Go-idiomatic way to copy a struct?

File: txnbuild/fee_bump_test.go [hunk 2]
```diff
@@ -191,6 +192,161 @@ func TestFeeBumpAllowsFeeAccountToEqualInnerSourceAccount(t *testing.T) {
...
+	assert.NoError(t, err)
+	feeBumpTx, err = feeBumpTx.Sign(network.TestNetworkPassphrase, kp1)
+	assert.NoError(t, err)
+	expectedBase64, err := feeBumpTx.Base64()
+	assert.NoError(t, err)
+
+	feeBumpTx, err = NewFeeBumpTransaction(
+		FeeBumpTransactionParams{
+			FeeAccount: kp1.Address(),
+			BaseFee:    2 * MinBaseFee,
```
[the remark this answers is not in the document]
Comment (tamirms) [reply txnbuild/fee_bump_test.go:227]: yes, that's correct. the fee bump transaction has an extra cost which is equivalent to adding another operation to the transaction, see https://github.com/stellar/stellar-protocol/blob/master/core/cap-0015.md#fee-rate

File: txnbuild/transaction.go (hunk 1 above)
Comment (tamirms) [reply txnbuild/transaction.go:498]: in Go the assignment operator copies by value. However, the assignment operator does a shallow copy. If you have any pointers or slices in your structs, the new struct will still have references to those fields. Here's an article which provides more details on this topic https://flaviocopes.com/go-copying-structs/

In this particular case, the signatures field is the only mutable field in the struct. There are no exported functions on  `FeeBumpTransaction` or `Transaction` which modify any of the other fields in the struct. The code creates a copy of the `FeeBumpTransaction` struct via assignment and then it does a deep copy on the signatures slice.

File: txnbuild/fee_bump_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply txnbuild/fee_bump_test.go:227]: Cool, thanks for sharing the CAP-15 reference.

**Pull request merged**